Mage.Sets/src/mage/cards/p/PactWeapon.java

```diff
@@ -0,0 +1,107 @@
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class PactWeapon extends CardImpl {
+
+    private static final Condition condition = new AttachedToMatchesFilterCondition(StaticFilters.FILTER_PERMANENT_CREATURE);
+
+    public PactWeapon(UUID ownerID, CardSetInfo setInfo) {
+        super(ownerID, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}{B}");
+
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // As long as Pact Weapon is attached to a creature, you don't lose the game for having 0 or less life.
+        this.addAbility(new SimpleStaticAbility(new ConditionalContinuousEffect(new DontLoseByZeroOrLessLifeEffect(Duration.WhileOnBattlefield), condition,
+                "as long as {this} is attached to a creature, you don't lose the game for having 0 or less life")));
+
+        // Whenever equipped creature attacks, draw a card and reveal it. The creature gets +X/+X until end of turn and you lose X life, where X is that card's mana value.
+        this.addAbility(new AttacksAttachedTriggeredAbility(new PactWeaponEffect()));
+
+        // Equip—Discard a card.
+        this.addAbility(new EquipAbility(Outcome.AddAbility, new DiscardCardCost(), false));
+    }
+
+    private PactWeapon(final PactWeapon card) {
+        super(card);
+    }
+
+    @Override
+    public PactWeapon copy() {
+        return new PactWeapon(this);
+    }
+}
+
+class PactWeaponEffect extends OneShotEffect {
+
+    public PactWeaponEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "draw a card and reveal it. The creature gets +X/+X until end of turn and you lose X life, where X is that card's mana value.";
+    }
+
+    private PactWeaponEffect(final PactWeaponEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PactWeaponEffect copy() {
+        return new PactWeaponEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        Permanent permanent = source.getSourcePermanentOrLKI(game);
+        if (permanent == null || game.getPermanent(permanent.getAttachedTo()) == null) {
+            return false;
+        }
+        Card card = controller.getLibrary().getFromTop(game);
+        // Gatherer ruling (2007-02-01)
+        // If the draw is replaced by another effect, none of the rest of Fa’adiyah Seer’s ability applies,
+        // even if the draw is replaced by another draw (such as with Enduring Renewal).
+        if (controller.drawCards(1, source, game) != 1) {
+            return true;
+        }
+        controller.revealCards(source, new CardsImpl(card), game);
+        if (card == null || card.getManaValue() == 0) {
+            return true;
+        }
```

Comment:
Shouldn't this still continue even if the manaValue is 0 incase there are replacement effects that increase the amount of life a players loses of the +X/+X they get?

Reply:
I believe that effects which increase the damage dealt by a source require the damage to at least be 1 or else the damage increase does not happen (e.g. with a [Torbran, Thane of Red Fell](https://scryfall.com/card/eld/147/torbran-thane-of-red-fell) in play, a hypothetical red instant reading "deal 0 damage to any target" would not have its damage increased  to 2, as no damage is being dealt in the first place), and though I don't believe any cards currently exist that have the second effect you described, I imagine they would also require at least a +1/+1 increase for it to increase anything (though it's hard to say as I can't think of any cards which do something like this and don't believe there's any ruling on it as a result)

Reply:
Alright. So I found a source for the 0 damage:
> 120.8. If a source would deal 0 damage, it does not deal damage at all. That means abilities that trigger on damage being dealt won’t trigger. It also means that replacement effects that would increase the damage dealt by that source, or would have that source deal that damage to a different object or player, have no event to replace, so they have no effect.

I couldn't find anything for +0/+0. I think it's like you said, no effect may interact with that.